cmd/joy/release.go

```diff
@@ -258,3 +266,111 @@ func NewReleaseRenderCmd() *cobra.Command {
 	cmd.Flags().BoolVar(&color, "color", term.IsTerminal(int(os.Stdout.Fd())), "toggle output with color")
 	return cmd
 }
+
+func NewGitQueryCommand(command string) *cobra.Command {
+	var (
+		source string
+		target string
+	)
+	cmd := &cobra.Command{
+		Use:  "git-" + command,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg := config.FromContext(cmd.Context())
+
+			cat, err := catalog.Load(catalog.LoadOpts{Dir: cfg.CatalogDir})
+			if err != nil {
+				return err
+			}
+
+			release := args[0]
+
+			var sourceRelease, targetRelease *v1alpha1.Release
+
+			for _, cross := range cat.Releases.Items {
+				if cross.Name != release {
+					continue
+				}
+				for _, rel := range cross.Releases {
+					if rel == nil {
+						continue
+					}
+					switch rel.Environment.Name {
+					case source:
+						sourceRelease = rel
+					case target:
+						targetRelease = rel
+					}
+				}
+				break
+			}
+
+			if sourceRelease == nil {
+				return fmt.Errorf("no source found")
+			}
+			if targetRelease == nil {
+				return fmt.Errorf("no target found")
+			}
+
+			repoCache := filepath.Join(cfg.JoyCache, "src")
```

Comment:
My first thought would have been to use the existing repo, but I agree having them standalone simplifies things.

Reply:
We can't really have people put their local configurations in the catalog, or assume their folder/repo-cloning conventions. So yeah! Disk space is free! maybe?